### ISSUE TITLE
odd clip rate from xml adaptor

### ISSUE THREAD

**Issue by alatdneg** [PixarAnimationStudios/OpenTimelineIO#334]
Is this right ? As it's breaking other code in otio as it's an invalid time rate ?

23.976023976

Should it not round it to 23.976?

This is triggered by this in the xml

<rate>
<timebase>24</timebase>
<ntsc>TRUE</ntsc>
</rate>

**Comment by alatdneg** [comment]
ValueError: 23.976023976 is not a valid frame rate, Please use one of these: (1, 12, 23.976, 23.98, 24, 25, 30, 48, 50, 60, 29.97, 59.94)
Should this validation be smarter ?

**Comment by reinecke** [comment]
The FCP XML adapter may be using sub-optimal parsing.
In the raw XML, 23.976 is specified as:
```
<rate>
    <timebase>24</timebase>
    <ntsc>TRUE</ntsc>
</rate>
```

And the adapter parses it using this function:
```
def _parse_rate(elem, element_map):
    elem = _resolved_backreference(elem, 'all_elements', element_map)

    rate = elem.find('./rate')
    # rate is encoded as a timebase (int) which can be drop-frame
    base = float(rate.find('./timebase').text)
    if rate.find('./ntsc').text == 'TRUE':
        base *= .999
    return base
```

This table from apple lays out some expectations: https://developer.apple.com/library/archive/documentation/AppleApplications/Reference/FinalCutPro_XML/FrameRate/FrameRate.html#//apple_ref/doc/uid/TP30001158-BCIHDFGD

But I think it's imperfect because it's still saying the "actual rate" is a decimal rather than a fraction like 24000/1001. However using this kind of lookup would likely be a bit more internally consistent with the OTIO timecode handling and if the same lookup is used on export it will be symmetric.

**Comment by thecargocultnz** [comment]
Yes, that "0.999" is an approximation, and will definitely cause rounding issues and drift over time.
The only way to correctly calculate NSTC values is using *1000/1001 (or  the inverse).
